### PR TITLE
Improve Fp16 tutorial and fix issues in scripts

### PR DESCRIPTION
**Pull request by rahul003**
## Description ##
- Improve the float16 tutorial by adding a point of comparision with larger batch size. 
- Fix a typo in the benchmark_score script, an issue with the gluon example
- Update fine-tune script after recent augmentation PR changed the way the augmentations are applied in that script.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by xinyu-intel** [comment]
@szha @rahul003 Please take a review and this can be a hotfix for `benchmark_score.py`. Thanks!

**Comment by rahul003** [comment]
@indhub Could you take a look?

File: docs/faq/float16.md [hunk 1]
```diff
@@ -102,9 +102,17 @@ python fine-tune.py --network resnet --num-layers 50 --pretrained-model imagenet
...
 
-![Training curves of Resnet50 v1 on Imagenet 2012](https://raw.githubusercontent.com/rahul003/web-data/03929a8beb8ac574f2392ed34cc6d4b2f052826a/mxnet/tutorials/mixed-precision/resnet50v1b_imagenet_fp16_fp32_training.png)
+Batch size | Data type | Top 1 Validation accuracy | Time to train | Speedup |
+--- | --- | --- | --- | --- |
+1024 | float32 | 76.18% | 11.8 hrs | 1 |
+1024 | float16 | 76.34% | 7.3 hrs | 1.62x |
+2048 | float16 | 76.29% | 6.5 hrs | 1.82x |
+
+![Training curves of Resnet50 v1 on Imagenet 2012](https://github.com/rahul003/web-data/blob/d415abf4a1c6df007483169c81807c250135f9a5/mxnet/tutorials/mixed-precision/resnet50v1b_imagenet_fp16_fp32_training.png?raw=true)
```
Comment (eric-haibin-lin) [inline docs/faq/float16.md]: Can we not use personal repo for images?

File: docs/faq/float16.md [hunk 2]
```diff
@@ -102,9 +102,17 @@ python fine-tune.py --network resnet --num-layers 50 --pretrained-model imagenet
...
+Let us consider training a Resnet50 v1 model on the Imagenet 2012 dataset. For this model, the GPU memory usage is close to the capacity of V100 GPU with a batch size of 128 when using float32. Using float16 allows the use of 256 batch size. Shared below are results using 8 V100 GPUs. Let us compare the three scenarios that arise here: float32 with 1024 batch size, float16 with 1024 batch size and float16 with 2048 batch size. These jobs trained for 90 epochs using a learning rate of 0.4 for 1024 batch size and 0.8 for 2048 batch size. This learning rate was decayed by a factor of 0.1 at the 30th, 60th and 80th epochs. The only changes made for the float16 jobs when compared to the float32 job were that the network and data were cast to float16, and the multi-precision mode was used for optimizer. The final accuracy at 90th epoch and the time to train are tabulated below for these three scenarios. The top-1 validation errors at the end of each epoch are also plotted below.
```
Comment (eric-haibin-lin) [inline docs/faq/float16.md]: It's better to be specific on the overall hardware setup (it's not done on DGX).

`Shared below are results using 8 V100 GPUs` ->
`Shared below are results using 8 V100 GPUs on AWS p3.16xlarge instance. `

File: example/gluon/data.py [hunk 3]
```diff
@@ -109,18 +111,17 @@ def transform(image, label):
         # resize the shorter edge to 224, the longer edge will be greater or equal to 224
         resized = mx.image.resize_short(image, 224)
         # center and crop an area of size (224,224)
-        cropped, crop_info = mx.image.center_crop(resized, 224)
+        cropped, crop_info = mx.image.center_crop(resized, (224, 224))
         # transpose the channels to be (3,224,224)
-        transposed = nd.transpose(cropped, (2, 0, 1))
-        image = mx.nd.cast(image, dtype)
-        return image, label
+        transposed = mx.nd.transpose(cropped, (2, 0, 1))
```
Comment (eric-haibin-lin) [inline example/gluon/data.py]: is dtype casting no longer necessary?

Comment (rahul003) [reply example/gluon/data.py]: Not in this script as it does astype() later. And that will also be more general for any dataset iterator

File: docs/faq/float16.md (hunk 2 above)
Comment (rahul003) [reply docs/faq/float16.md]: Ok yeah

File: docs/faq/float16.md (hunk 1 above)
Comment (rahul003) [reply docs/faq/float16.md]: Unfortunately I have a hard time getting anything merged into DMLC repos. Looks like there have been more than 25 commits since I opened this PR 4 weeks back there https://github.com/dmlc/web-data/pull/79 but this is still unnoticed!

**Comment by rahul003** [comment]
@hetong007 please confirm the resnet-aug change to turn on random_mirror.

**Comment by rahul003** [comment]
@eric-haibin-lin The training curve image is now in dmlc repository.

**Comment by hetong007** [comment]
For cifar training, the standard augmentation for benchmark is:

```python
mean_rgb = [125.307, 122.961, 113.8575]
std_rgb = [51.5865, 50.847, 51.255]
train_data = mx.io.ImageRecordIter(
    path_imgrec         = rec_train,
    path_imgidx         = rec_train_idx,
    preprocess_threads  = num_workers,
    shuffle             = True,
    batch_size          = batch_size,
    
    data_shape          = (3, 32, 32),
    mean_r              = mean_rgb[0],
    mean_g              = mean_rgb[1],
    mean_b              = mean_rgb[2],
    std_r               = std_rgb[0],
    std_g               = std_rgb[1],
    std_b               = std_rgb[2],
    rand_mirror         = True,
    pad                 = 4,
    fill_value          = 0,
    rand_crop           = True,
    max_crop_size       = 32,
    min_crop_size       = 32,
)
val_data = mx.io.ImageRecordIter(
    path_imgrec         = rec_val,
    path_imgidx         = rec_val_idx,
    preprocess_threads  = num_workers,
    shuffle             = False,
    batch_size          = batch_size,
    
    data_shape          = (3, 32, 32),
    mean_r              = mean_rgb[0],
    mean_g              = mean_rgb[1],
    mean_b              = mean_rgb[2],
    std_r               = std_rgb[0],
    std_g               = std_rgb[1],
    std_b               = std_rgb[2],
)
```

Can you change accordingly?

**Comment by rahul003** [comment]
Are the above augmentations used for all Cifar models?

**Comment by hetong007** [comment]
@rahul003 I think it is more appropriate to call `set_imagenet_aug` as it applies to a bunch of popular models.

And yes the above params are standard for cifar model training & performance comparison.

File: example/image-classification/train_cifar10.py [hunk 4]
```diff
@@ -31,6 +31,12 @@ def download_cifar10():
     download_file('http://data.mxnet.io/data/cifar10/cifar10_train.rec', fnames[0])
     return fnames
 
+def set_cifar_aug(aug):
+    aug.set_defaults(rgb_mean='125.307,122.961,113.8575', rgb_std='51.5865,50.847,51.255')
+    aug.set_defaults(random_mirror=1, pad=4, fill_value=0, random_crop=1)
+    aug.set_defaults(min_random_size=32, max_random_size=32)
+
+
```
Comment (hetong007) [inline example/image-classification/train_cifar10.py]: a few blank lines (also in `train_imagenet.py`:L26-27), otherwise lgtm.

**Comment by rahul003** [comment]
@eric-haibin-lin @hetong007 Could you merge this?
